### PR TITLE
KOGITO-5636 Native build image with another image

### PR DESCRIPTION
**Pull request by radtriste**
https://issues.redhat.com/browse/KOGITO-5636

Many thanks for submitting your Pull Request :heart:! 

Please make sure your PR meets the following requirements:

- [x] You have read the [contributors' guide](https://github.com/kiegroup/kogito-operator/blob/main/README.md#contributing-to-the-kogito-operator)
- [x] Pull Request title is properly formatted: `[KOGITO-XYZ] Subject`
- [x] Pull Request contains a link to the JIRA issue
- [x] Pull Request contains a description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket
- [x] Your feature/bug fix has a unit test that verifies it
- [ ] You've ran `make before-pr` and everything is working accordingly
- [x] You've tested the new feature/bug fix in an actual OpenShift cluster
- [ ] You've added a [RELEASE_NOTES.md](https://github.com/kiegroup/kogito-operator/blob/main/RELEASE_NOTES.md) entry regarding this change

### CODE REVIEW

File: test/features/deploy_kogito_build.feature [hunk 1]
```diff
@@ -40,42 +41,6 @@ Feature: Deploy Kogito Build
     And Local example service "<example-service>" is built by Maven with configuration:
       | native | <native> |
 
-    When Build binary <runtime> service "<example-service>" with configuration:
```
Comment (sutaakar) [inline test/features/deploy_kogito_build.feature:43]: Why did you delete this test?

Comment (radtriste) [reply test/features/deploy_kogito_build.feature:43]: `Build <runtime> binary build with native <native> using KogitoBuild` and `Build <runtime> binary build from local example service target folder with native <native> using KogitoBuild` are performing the test AFAIU
the first one was set before the CLI could handle the upload of binaries, see https://github.com/kiegroup/kogito-operator/pull/976/files?file-filters%5B%5D=.bats&file-filters%5B%5D=.deploy&file-filters%5B%5D=.feature&file-filters%5B%5D=.go&file-filters%5B%5D=.groovy&file-filters%5B%5D=.sh&file-filters%5B%5D=No+extension#diff-aab0846fc8b6246b32aca272924400d2dd8c8306e0dc2d536a0a6c94e4162ce3L47

Comment (sutaakar) [reply test/features/deploy_kogito_build.feature:43]: ok, thanks

**Review by sutaakar — APPROVED** [review]
